File: storage/data_transpiled/at_coder/arc098/B/3208575.java
public static int N ( ) {
  int N = Integer . parseInt ( input ( ) ) ;
  int [ ] A = Lists . newArrayList ( ) ;
  for ( int i = 0 ;
  i < N ;
  i ++ ) A [ i ] = Integer . parseInt ( input ( ) ) ;
  int ans = 0 , r = 0 ;
  int XOR = 0 , SUM = 0 ;
  for ( int l = 0 ;
  l < N ;
  l ++ ) {
    while ( r < N && XOR ^ A [ r ] == SUM + A [ r ] ) {
      XOR ^= A [ r ] ;
      SUM += A [ r ] ;
      r ++ ;
    }
    ans += r - l ;
    XOR ^= A [ l ] ;
    SUM -= A [ l ] ;
  }
  System . out . println ( ans ) ;
  return ans ;
}

File: data/data_original/at_coder/arc098/B/3208575.py
N = int(input())
A = list(map(int, input().split()))

ans, r = 0, 0
XOR, SUM = 0, 0
for l in range(N):
    while r < N and XOR ^ A[r] == SUM + A[r]:
        XOR ^= A[r]
        SUM += A[r]
        r += 1

    ans += r - l
    XOR ^= A[l]
    SUM -= A[l]

print(ans)